### PR TITLE
Add missing overrides for ModifierVisitorAdapter

### LINKED ISSUE #300: ModifierVisitorAdapter does not provide default overrides for UnionType and IntersectionType
I'll try to volunteer a patch later this week.

### PR DESCRIPTION
**Pull request by thejohnfreeman**
Closes #300

### CODE REVIEW

**Comment by ftomassetti** [comment]
It seems to not compile:

```
[ERROR] /home/travis/build/javaparser/javaparser/javaparser-core/src/main/java/com/github/javaparser/ast/visitor/ModifierVisitorAdapter.java:[804,12] cannot find symbol
  symbol:   class R
  location: class com.github.javaparser.ast.visitor.ModifierVisitorAdapter<A>
```

**Comment by ftomassetti** [comment]
I think it makes sense but we should fix the return type of the second method

**Comment by thejohnfreeman** [comment]
Yeah, sorry about that. I fixed that and tried to add a test. Let me know if it doesn't meet the standards or is in the wrong place. This is my first time using JBehave.

**Comment by ftomassetti** [comment]
No problem, tests are there to help us catch these things :)

File: javaparser-testing/src/test/java/com/github/javaparser/ast/visitor/ModifierVisitorAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+package com.github.javaparser.ast.visitor;
+
+import org.jbehave.core.annotations.Then;
+
+import static org.junit.Assert.assertNotNull;
+
+public class ModifierVisitorAdapterTest {
+
+  @Then("ModifierVisitorAdapter requires no overrides")
```
Comment (ftomassetti) [inline javaparser-testing/src/test/java/com/github/javaparser/ast/visitor/ModifierVisitorAdapterTest.java]: I do not understand what this test is doing. It instantiate an object and then it verifies it is not null?

**Comment by thejohnfreeman** [comment]
Yes. The real test is that the construction compiles. If ModifierVisitorAdapter is missing any overrides like it was before this patch, then the construction will not compile. I added the assertion just so that the test has at least one. I'll make this test look however you want, just tell me.

**Comment by ftomassetti** [comment]
I would suggest we just remove the test and merge the change as it is.

Thanks for your help on this one!

**Comment by thejohnfreeman** [comment]
As you wish.

**Comment by ftomassetti** [comment]
Looks perfect. Thanks again!

**Pull request merged**